File: DSA/patterns/frequency_counter/CharacterFrequency.js
// CharacterFrequency.js

function characterFrequency(s) {
  /**
   * Count the frequency of each character in a string using the Frequency Counter pattern.
   */
  let freqCount = {};

  for (let char of s) {
    freqCount[char] = (freqCount[char] || 0) + 1;
  }

  return freqCount;
}

File: DSA/patterns/frequency_counter/character_frequency.py
"""character_frequency.py"""

def character_frequency(s):
    """
    Count the frequency of each character in a string using the Frequency Counter pattern.
    """
    freq_count = {}

    for char in s:
        freq_count[char] = freq_count.get(char, 0) + 1

    return freq_count
